fem/datacollection.cpp

```diff
@@ -726,4 +726,392 @@ void VisItDataCollection::ParseVisItRootString(const std::string& json)
+   //define the vtu file
+   {
+      std::string fname=GenerateCollectionPath()+"/"+GenerateVTUPath()+"/"
+                        +GenerateVTUFileName();
+      std::fstream out; out.open(fname.c_str(), std::ios::out);
+      SaveDataVTU(out,levels_of_detail);
+      out.close();
+   }
+   //define the pvtu file only on process 0
+   if (myrank==0)
+   {
+      std::string fname=GenerateCollectionPath()+"/"+GeneratePVTUPath()+"/"
+                        +GeneratePVTUFileName();
+      std::fstream out; out.open(fname.c_str(), std::ios::out);
+
+      out<< "<?xml version=\"1.0\"?>" << std::endl;
+      out<< "<VTKFile type=\"PUnstructuredGrid\" version =\"0.1\" byte_order=\"LittleEndian\"> "
+         << std::endl;
+      out<< "<PUnstructuredGrid GhostLevel=\"0\">" << std::endl ;
+
+      out<< "<PPoints>" << std::endl;
+      out<< "\t<PDataArray type=\"Float32\" Name=\"Points\" NumberOfComponents=\"3\"/>"
+         << std::endl;
+      out<< "</PPoints>" << std::endl;
+
+      out<< "<PCells>" << std::endl ;
+      out<< "\t<PDataArray type=\"Int32\" Name=\"connectivity\" NumberOfComponents=\"1\"/>"
+         << std::endl ;
+      out<< "\t<PDataArray type=\"Int32\" Name=\"offsets\"      NumberOfComponents=\"1\"/>"
+         << std::endl ;
+      out<< "\t<PDataArray type=\"UInt8\" Name=\"types\"        NumberOfComponents=\"1\"/>"
+         << std::endl ;
+      out<< "</PCells>" << std::endl ;
+
+
+      out<< "<PPointData>" << std::endl ;
+      for (FieldMapIterator it=field_map.begin(); it!=field_map.end(); ++it)
+      {
+         out<<"<PDataArray type=\"Float32\" Name=\""<<it->first;
+         int vec_dim=it->second->VectorDim();
+         out<<"\" NumberOfComponents=\""<<vec_dim<<"\" format=\"ascii\" />"<<std::endl;
+      }
+      out<<"</PPointData>" << std::endl ;
+
+      //CELL DATA
+      out<< "<PCellData>" << std::endl ;
+      out<< "\t<PDataArray type=\"Int32\" Name=\"" << "material"
+         <<"\" NumberOfComponents=\"1\"/> " << std::endl ;
+      out<< "\t<PDataArray type=\"Int32\" Name=\"" << "element_coloring"
+         <<"\" NumberOfComponents=\"1\"/> " << std::endl ;
+      out<< "</PCellData>" << std::endl ;
+
+      for (int ii=0; ii<nprocs; ii++)
+      {
+         //this one is generated without the path
+         std::string nfname=GenerateVTUFileName(ii);
+         out<<"<Piece Source=\""<<nfname<<"\"/>"<<std::endl;
+      }
+      out<<"</PUnstructuredGrid>"<<std::endl;
+      out<<"</VTKFile>"<<std::endl;
+      out.close();
+
+
+      fname=GeneratePVTUPath()+"/"+GeneratePVTUFileName();
+      //add the pvtu file to the pvd_stream
+      pvd_stream<<"<DataSet timestep=\""<<GetTime();  //GetCycle();
+      pvd_stream<<"\" group=\"\" part=\""<<0<<"\" file=\"";
+      pvd_stream<<fname<<"\"/>"<<std::endl;
+   }
+
+
+}
+
+
+
+void ParaviewDataCollection::SaveDataVTU(std::ostream &out, int ref)
+{
+   out<<"<VTKFile type=\"UnstructuredGrid\" version=\"0.1\" byte_order=\"LittleEndian\">"<<std::endl;
```

Comment:
Running `make style` does not break this long line since it has no spaces. Adding spaces manually in this and other lines allows `make style` to format them.

Reply:
Should be fine in the last version.